### DISCUSSION TITLE
Getting Back to Factory Defaults

### DISCUSSION THREAD

**Discussion by erentkll** [polarfire-soc/polarfire-soc-documentation#326] [HSS and U-Boot] [answered]
Hi, I want to get back to factory defaults for my PolarFire SoC. I have watched the video: https://youtu.be/Foil5pIRBlM
There is no Linux in the board currently, so I want to upload a Linux image. However, when I upload the .wic file, HSS cannot start the Linux. Do I need to upgrade my HSS because its version is very old. If so, can I install FlashPro Express without using Libero for flashing the .job file?

**Comment by hughbreslin** [comment, marked as answer]
Hi @erentkll You probably need to update your HSS - we do provide standalone FlashPro Express as part of our "Program and Debug Tools" but I'm actually struggling to find the download button! This is the [page](https://www.microchip.com/en-us/products/fpgas-and-plds/fpga-and-soc-design-tools/programming-and-debug) you're looking for and I'll comment here with the link to download the tools once I get it. In the meantime you could install Libero and this bundles FlashPro Express so you'll get FPExpress with the Libero install without having to use Libero.

**Reply by erentkll** [reply]
Thank you for your reply. I fixed the problem by using a Linux image that is suitable for my HSS version which is very old :) Still, I have to update the current version by considering your explanation. Thank you.

**Reply by hughbreslin** [reply]
@erentkll here are the direct links to download:

[Windows](https://ww1.microchip.com/downloads/secure/aemdocuments/documents/fpga/media-content/fpga/v2023-1/program_debug_v2023.1_win.exe)
[Linux](https://ww1.microchip.com/downloads/secure/aemdocuments/documents/fpga/media-content/fpga/v2023-1/program_debug_v2023.1_lin.zip)
[Release notes](https://coredocs.s3.amazonaws.com/Libero/2023_1/Tool/programming_debug_v2023_1_release_notes.pdf)

This will let you set up program and debug (i.e a standalone install of FPExpress without Libero) :)

**Reply by erentkll** [reply]
@hughbreslin thank you for your support. I will try to update the HSS. Have a nice day!
